Calculator/Calculator/Program.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+
+namespace Calculator
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int previous;
+            int current;
+            string operation;
+
+            while(true)
+            {
+                StartCalculation:
+                while(true)
+                {
+                    Console.Write("Введите число: ");
+                    var priv = Console.ReadLine();
+                    try
+                    {
+                        previous = int.Parse(priv);
+                        break;
+                    }
+                    catch (Exception)
+                    {
+                        Console.WriteLine("Это не число");
+                    }
+                }
+
+                while(true)
+                {
+                    Console.Write("Введите операцию: ");
+                    operation = Console.ReadLine();
+                    if (operation == "!")
+                    {
+                        Console.WriteLine($"Факториал числа {previous} равен {Factorial(previous)}");
+                        goto StartCalculation;
+                    }
+                    if (operation == "+" || operation == "-" || operation == "*" || operation == "/")
```

Comment:
You forgot about "Exponentiation"

Reply:
Agreed, added exponentiation logic in commit 4732b8dfedbf698a85e75e01c5123f589472f429